### DISCUSSION TITLE
About DocumentReference Type

### DISCUSSION THREAD

**Discussion by mysCod3r** [firebase/flutterfire#9810] [Q&A]
Hi there,
I have reference type in firestore document and a model class.
I am accessing this data in Flutter application. `DocumentSnapshot<Object?>?` I get this type of return. What should I do to convert it to a class I wrote instead of an Object?    -> like this --> `DocumentSnapshot<UserModel?>?`

![My PostModel class](https://user-images.githubusercontent.com/76634747/198411726-b185f210-b07a-4926-bdc3-1537d8d74655.jpg)

![My UserModel](https://user-images.githubusercontent.com/76634747/198412036-a55ea2a4-1042-4440-a7db-0de797b81251.jpg)

![My Code](https://user-images.githubusercontent.com/76634747/198412679-2289abf1-308c-40c0-b620-6bd20d63e9d9.jpg)

I tried to add cast but it didn't work.

**Comment by Ehesp** [comment]
You can use the `withConverter` API for that: https://pub.dev/documentation/cloud_firestore/latest/cloud_firestore/Query/withConverter.html

```dart
.collection('persons')
    .where('age', isGreaterThan: 0)
    .withConverter<Person>(
      fromFirestore: (snapshot, _) => Person.fromJson(snapshot.data()!),
      toFirestore: (person, _) => person.toJson(),
    );
```
